### PR TITLE
Replace restricted identifier 'yield' usage

### PR DESCRIPTION
**Pull request by korthout**
## Description

Since JDK 14, `yield` has become a restricted identifier.

## Related issues

See: openjdk.java.net/jeps/361

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: util/src/main/java/io/camunda/zeebe/util/sched/ActorControl.java [hunk 1]
```diff
@@ -334,7 +335,7 @@ private TimerSubscription scheduleTimer(
   }
 
   /** can be called by the actor to yield the thread */
-  public void yield() {
+  public void yieldThread() {
     final ActorJob job = ensureCalledFromWithinActor("yield()");
```
Comment (korthout) [inline util/src/main/java/io/camunda/zeebe/util/sched/ActorControl.java:339]: I was unsure about it. Should we also change the name on this line?

Comment (korthout) [inline util/src/main/java/io/camunda/zeebe/util/sched/ActorControl.java:339]: I was unsure about it. Should we also change the name on this line?

Comment (npepinpe) [reply util/src/main/java/io/camunda/zeebe/util/sched/ActorControl.java:339]: Yes, let's avoid confusion there.

**Review by npepinpe — APPROVED** [review]
bors merge

:+1:

**Review by npepinpe — APPROVED** [review]
bors merge

:+1:

**Comment by korthout** [comment]
> Build failed:
> 
> * [continuous-integration/jenkins/branch](https://ci.zeebe.camunda.cloud/job/camunda-cloud/job/zeebe/job/staging/880/display/redirect)

Flaky test

bors retry

**Comment by korthout** [comment]
bors retry

**Pull request merged**